Client/InfluxDBClient.cs

```diff
@@ -457,19 +458,20 @@ public InfluxDBClient() : this(
             await foreach (var batch in QueryBatches(query, queryType, database, namedParameters, headers)
                                .ConfigureAwait(false))
             {
-                var rowCount = batch.Column(0).Length;
-                for (var i = 0; i < rowCount; i++)
+                for (var i = 0; i < batch.Column(0).Length; i++)
                 {
-                    var row = new List<object?>();
-                    for (var j = 0; j < batch.ColumnCount; j++)
+                    var columnCount = batch.ColumnCount;
+                    var row = new object?[columnCount];
+                    for (var j = 0; j < columnCount; j++)
                     {
-                        if (batch.Column(j) is ArrowArray array)
-                        {
-                            row.Add(array.GetObjectValue(i));
-                        }
+                        if (batch.Column(j) is not ArrowArray array) continue;
```

Comment:
Use braces for better readability:
```suggestion
                        if (batch.Column(j) is not ArrowArray array) 
                        {
                                    continue;
                        }
```